src/Storages/ExternalStream/StorageExternalStream.cpp

```diff
@@ -39,6 +39,9 @@ buildShardingKeyExpression(ASTPtr sharding_key, ContextPtr context, const NamesA
 }
 
 void validateEngineArgs(ContextPtr context, ASTs & engine_args, const ColumnsDescription & columns) {
+    if (engine_args.size() == 0)
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "ExternalStream requires sharding key expression");
+
```

Comment:
A better way is that when no specified sharing key expression, use a default shading expression `rand()`

Reply:
resolved